modules/core/src/quaternion.cpp

```diff
@@ -18,74 +18,93 @@
 // Unless required by applicable law or agreed to in writing, software
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and 
-// limitations under the License. 
+// See the License for the specific language governing permissions and
+// limitations under the License.
 //
-// Author: Longbu Wang <riskiest@gmail.com>
-//         Liangqian Kong <chargerKong@126.com>:
-
+// Author: Liangqian Kong <chargerKong@126.com>
+//         Longbu Wang <riskiest@gmail.com>
 
 #include "precomp.hpp"
-#include "opencv2/core/quaternion.hpp"
-#define EPS 0.0001
 #include <vector>
-using namespace cv;
-template <typename T>
-Quat<T>::Quat(const cv::Vec<T, 4> &coeff):w(coeff[0]), x(coeff[1]), y(coeff[2]), z(coeff[3]){}
 
+namespace cv{
+template <typename T>
+Quat<T>::Quat(const Vec<T, 4> &coeff):w(coeff[0]), x(coeff[1]), y(coeff[2]), z(coeff[3]){}
 
 template <typename T>
 Quat<T>::Quat(const T qw, const T qx, const T qy, const T qz):w(qw), x(qx), y(qy), z(qz){}
 
 template <typename T>
-Quat<T>::Quat(const T angle, const cv::Vec<T, 3> &axis)
-{ 	
-    T vNorm = cv::sqrt(axis.dot(axis));
-    w = std::cos(angle / 2);
-    x = std::sin(angle / 2) * (axis[0] / vNorm);
-    y = std::sin(angle / 2) * (axis[1] / vNorm);
-    z = std::sin(angle / 2) * (axis[2] / vNorm);
+Quat<T>::Quat(const T angle, const cv::Vec<T, 3> &axis, const T qNorm)
```

Comment:
Do we need a separate `qNorm` argument? The same effect can be achieved by multiplying a quaternion by a number.

Reply:
Thanks for your code review. 

Rerferring to the definition of a quaternion in [wikipedia],(https://en.wikipedia.org/wiki/Quaternion) 

![image](https://user-images.githubusercontent.com/49099366/94359623-1714a980-00db-11eb-9371-77bfb6e80ea5.png)


So I think a quaternion is constructed by this three params. But generally speaking, 

when we set an angle and an axis of a quat, it is a unit quat by default. That's why I

set the default value for qNorm. 


maybe we could save some computations by

```
if (qNorm == 1)
{
    w = std::cos(angle / 2);
    ...
}
else
{
    w = std::cos(angle / 2) * qNorm;
    ....
} 
```

Reply:
No, having an `if(qNorm == 1)` won't save any performance I guess, a condition takes some clocks too.

I mean if a user wants to create an arbitrary norm quaternion, there's an intuitive way for that:
```
Quat<float> theQuat = Quat<float>(theAngle, theAxis) * theNorm;
```
And this way does not require that 3-argument constructor, just 2-argument, for axis-angle.

Reply:
OK, thanks very much for your suggestion. We will delete it later.